scripts/generate-integration-pages.ts

```diff
@@ -0,0 +1,177 @@
+			.use(relativeLinks, { base: `https://docs.astro.build/` })
+			.use(githubVideos);
+		readme = (await processor.process(readme)).toString();
+		readme =
+			`---
+# NOTE: This file is auto-generated from 'scripts/generate-integration-pages.ts'
+#       and pulls content directly from the package’s README.
+#       DO NOT MAKE EDITS TO THIS FILE DIRECTLY, THEY WILL BE OVERWRITTEN!
+#       For corrections, please edit the package README at
+#       ${githubLink}
+
+layout: ~/layouts/IntegrationLayout.astro
+title: '${name}'
+version: '${version}'
+githubURL: '${githubLink}'
+category: ${category}
+i18nReady: false
+---\n\n` + readme;
+		return readme;
+	}
+
+	async #writeReadme(packageName: string, readme: string): Promise<void> {
+		const unscopedName = packageName.split('/').pop();
+		return await fs.promises.writeFile(`src/pages/en/guides/integrations-guide/${unscopedName}.md`, readme, 'utf8');
+	}
+
+	async run() {
+		console.log(kleur.bold('Creating integration pages...\n'));
+		const integrations = await this.#getIntegrationData();
+		await Promise.all(
+			integrations.map(async (integration) => {
+				const readme = await this.#processReadme(integration);
+				await this.#writeReadme(integration.name, readme);
+				console.log('', kleur.green('✔︎'), integration.name);
+			})
+		);
+		console.log('\n');
+	}
+}
+
+const builder = new IntegrationPagesBuilder({
+	sourceBranch: process.env.SOURCE_BRANCH || 'main',
+	sourceRepo: process.env.SOURCE_REPO || 'withastro/astro',
+	githubToken: process.env.GITHUB_TOKEN,
+});
+builder.run();
+
+/** Remark plugin to prepend an absolute path in front of link hrefs. */
+function absoluteLinks({ base }: { base: string }) {
+	return function transform(tree: Root) {
+		function visitor(node: Link | Definition) {
+			// Sanitize URL by removing leading `/`
+			const relativeUrl = node.url.replace(/^.?\//, '');
+			node.url = new URL(relativeUrl, base).href;
+		}
+		visit(tree, 'link', visitor);
+		visit(tree, 'definition', visitor);
+		visit(tree, 'html', function htmlVisitor(node) {
+			node.value = node.value.replace(/(?<=href=")(?!https?:\/\/)\/?(.+)(?=")/g, `${base}$1`);
+		});
+	};
+}
+
+/** Remark plugin to strip the docs base from absolute link hrefs. */
+function relativeLinks({ base }: { base: string }) {
+	return function transform(tree: Root) {
+		function visitor(node: Link | Definition) {
+			if (!node.url.startsWith(base)) return;
+			node.url = new URL(node.url).pathname;
+		}
+		visit(tree, 'link', visitor);
+		visit(tree, 'definition', visitor);
+		visit(tree, 'html', function htmlVisitor(node) {
+			node.value = node.value.replace(new RegExp(base, 'g'), '/');
+		});
+	};
+}
+
+/** Remark plugin to convert GitHub video URLs to `<video>` elements. */
+function githubVideos() {
+	return function transform(tree: Root) {
+		visit(tree, 'text', function visitor(node: Text | HTML) {
+			if (node.value.startsWith('https://user-images.githubusercontent.com/')) {
```

Comment:
Any idea how large the video files will be?  I think github caps out at 50MB, but that may be plenty for quick integration intro videos :+1:

Reply:
more of a note for the future really, hosting them in github works great and we can always change that later if the need arises!

Reply:
Good point! This was more to handle the existing videos in READMEs rather than ruling on future practice. The benefit of this format is that they play in-line on GitHub which isn’t possible hosting them elsewhere, but they _don’t_ work on npm, so it’s patchy. The most universal would probably be a poster image linked to a hosted video, which we could transform to an inline video on the docs site. We can cross that bridge if we feel like we need a lot more videos for docs.